File: Web/ExtranetApi/Models/ProvidersContainerDto.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace TravelLine.WebAppTemplate.ExtranetApi.Models
{
    [DataContract]
    public class ProvidersContainerDto
    {
        [DataMember( Name = "totalCount" )]
        public int TotalCount { get; set; }

        [DataMember( Name = "filteredCount" )]
        public int FilteredCount { get; set; }

        [DataMember( Name = "items" )]
        public List<ProviderDto> Items { get; set; }
    }
}

File: Web/Extranet/src/dto/provider/ProvidersContainerDto.ts
module WebAppTemplate.Dto {

    export class ProvidersContainerDto {
        private _totalCount: number;
        private _filteredCount: number;
        private _items: Dto.Provider[];

        constructor( totalCount: number, filteredCount: number, items: Dto.Provider[] ) {
            this._totalCount = totalCount;
            this._filteredCount = filteredCount;
            this._items = items;
        }

        public get totalCount(): number {
            return this._totalCount;
        }

        public get filteredCount(): number {
            return this._filteredCount;
        }

        public get items(): Dto.Provider[] {
            return this._items;
        }
    }
}